### PR TITLE
DayComics : get hidden comics

### PR DESCRIPTION
**Pull request by MikeZeDev**
Some comics requires coins on account / a purchase to be shown.

* Get comics from user library in case some are hidden ones.
* Enhance api requests to work on such content when user is logged. It affects pasting and getting chapters.
* Non logged users are not affected.

Closes https://github.com/manga-download/haruneko/issues/800

### CODE REVIEW

**Comment by ronny1982** [comment]
Isn't it possible to use a single request instead of two?
e.g., by optimising the injected script to directly extract the desired data via CSS or `await TopcoEpisodeRequestApi.EpisodeData(...)`.

**Comment by MikeZeDev** [comment]
> Isn't it possible to use a single request instead of two? e.g., by optimising the injected script to directly extract the desired data via CSS, accessing `__NEXT_DATA__` or `await webpackChunk_N_E[...]TopcoEpisodeRequestApi.EpisodeData(...)`.

Probably. Handling is a pain tho

**Comment by MikeZeDev** [comment]
* __NEXT_DATA__ is usable for clipboard (cause varaible is fetched in browser context so user would be logged and have access to hidden comics whom he pasted the link
* On Library, __NEXT_DATA__  is empty, meaning we have to use the api and therefore getting all those silly parameters

File: web/src/engine/websites/DayComics.ts [hunk 1]
```diff
@@ -38,34 +35,60 @@ const matureCookieScript = `
     });
 `;
 
+const libraryScript = `
+    [...document.querySelectorAll('section#libraryPage a[href*="/content/"')].map(anchor => {
+      const link = anchor.pathname;
```
Comment (ronny1982) [inline web/src/engine/websites/DayComics.ts:40]: no need for intermediate variable if they are only used once => assign the values directly in the return object

File: web/src/engine/websites/DayComics.ts [hunk 2]
```diff
@@ -38,34 +35,60 @@ const matureCookieScript = `
...
+      const title = anchor.querySelector('div.comicInfo > div > div').textContent.trim();
+      return {
+          id : link,
+          title : title
+      }});
+`;
+
+const chapterScript = `
+    [...document.querySelectorAll('a#episodeItemCon')].map(anchor => {
+      const link = anchor.pathname;
```
Comment (ronny1982) [inline web/src/engine/websites/DayComics.ts:50]: No need for intermediate variable `link`

**Review by ronny1982 — APPROVED** [review]

File: web/src/engine/websites/DayComics_e2e.ts [hunk 3]
```diff
@@ -8,12 +8,12 @@ const config = {
...
     container: {
         url: 'https://daycomics.com/content/101023',
-        id: '101023',
+        id: '/content/101023',
         title: 'Sex Study Group'
     },
     child: {
         id: '/content/101023/113522',
-        title: 'Episode 1'
+        title: 'Episode 1: Why is she in my house?'
```
[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/DayComics_e2e.ts:16]: Its intentional. I mean, it should have been included in the first place. Website display it after all.
Should i remove it?

File: web/src/engine/websites/DayComics.ts [hunk 4]
```diff
@@ -80,35 +103,35 @@ export default class extends DecoratableMangaScraper {
     }
 
     public override async Initialize(): Promise<void> {
-        return await FetchWindowScript(new Request(this.URI), matureCookieScript, 500);
+        return FetchWindowScript(new Request(this.URI), matureCookieScript, 500);
```
[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/DayComics.ts:106]: I know that. You may even not being able to browse the website at all if you get the cookies wrong (first when my code was wrong).

If you have a better idea to set +18 cookie ill take it.

File: web/src/engine/websites/DayComics_e2e.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (ronny1982) [reply web/src/engine/websites/DayComics_e2e.ts:16]: Depends on how popular the website is and the possibility that HakuNeko users have downloaded many chapters from this website 🤷‍♂️

[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/DayComics_e2e.ts:16]: It was introduced in Haruneko just a few weeks ago. It was not existing in Haruneko. 
I think its better adding that now than later, were more people may complain. I should have paid more attention to their api tho.

File: web/src/engine/websites/DayComics.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/DayComics.ts:97]: Perhaps we should give an option to clear all cookies is Settings ? Its more handy than opening Devtools.

[the remark this answers is not in the document]
Comment (MikeZeDev) [reply web/src/engine/websites/DayComics.ts:99]: Done. In case of error, non logged mature cookie is deleted. Promise always resolve.

**Pull request merged**